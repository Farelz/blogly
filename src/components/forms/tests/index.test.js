import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from '../login';

describe('forms tests', () => {
    const onSubmit = jest.fn();
    const onChange = jest.fn();
    const props = {
        onSubmit,
        onChange,
        loggingIn: false,
        auth_message: '',
        emailValue: '',
        passwordValue: ''
    };

    it('renders', () => {
        const wrapper = shallow(<LoginForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
