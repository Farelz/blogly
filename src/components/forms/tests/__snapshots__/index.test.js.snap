// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forms tests renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    auth_message=""
    emailValue=""
    loggingIn={false}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
    passwordValue=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-field col s12"
        >
          <input
            id="email"
            onChange={[MockFunction]}
            placeholder="Your email"
            type="text"
            value=""
          />
        </div>,
        <div
          className="input-field col s12"
        >
          <input
            id="password"
            onChange={[MockFunction]}
            placeholder="Your password"
            type="password"
            value=""
          />
        </div>,
        <div
          className="input-field col s12"
        >
          <button
            className="btn"
            type="submit"
          >
            Submit
          </button>
          <p
            className="red-text"
          >
            
          </p>
        </div>,
      ],
      "className": "row",
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            id="email"
            onChange={[MockFunction]}
            placeholder="Your email"
            type="text"
            value=""
          />,
          "className": "input-field col s12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "onChange": [MockFunction],
            "placeholder": "Your email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            id="password"
            onChange={[MockFunction]}
            placeholder="Your password"
            type="password"
            value=""
          />,
          "className": "input-field col s12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "onChange": [MockFunction],
            "placeholder": "Your password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn"
              type="submit"
            >
              Submit
            </button>,
            <p
              className="red-text"
            >
              
            </p>,
          ],
          "className": "input-field col s12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "red-text",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-field col s12"
          >
            <input
              id="email"
              onChange={[MockFunction]}
              placeholder="Your email"
              type="text"
              value=""
            />
          </div>,
          <div
            className="input-field col s12"
          >
            <input
              id="password"
              onChange={[MockFunction]}
              placeholder="Your password"
              type="password"
              value=""
            />
          </div>,
          <div
            className="input-field col s12"
          >
            <button
              className="btn"
              type="submit"
            >
              Submit
            </button>
            <p
              className="red-text"
            >
              
            </p>
          </div>,
        ],
        "className": "row",
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              id="email"
              onChange={[MockFunction]}
              placeholder="Your email"
              type="text"
              value=""
            />,
            "className": "input-field col s12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "onChange": [MockFunction],
              "placeholder": "Your email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              id="password"
              onChange={[MockFunction]}
              placeholder="Your password"
              type="password"
              value=""
            />,
            "className": "input-field col s12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "onChange": [MockFunction],
              "placeholder": "Your password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn"
                type="submit"
              >
                Submit
              </button>,
              <p
                className="red-text"
              >
                
              </p>,
            ],
            "className": "input-field col s12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "red-text",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
